api/src/controllers/post/inspection.js

```diff
@@ -0,0 +1,178 @@
+var mongoose = require('mongoose');
+var ObjectId = require('mongoose').Types.ObjectId;
+
+// Example of incomingObj
+/**
+ *    inpsections: [
+ *     {
+ *       recordName: 'test abc',
+ *       recordType: 'whatever',
+ *       ...
+ *       InspectionLNG: {
+ *          description: 'lng description'
+ *          addRole: 'public',
+ *       }
+ *     },
+ */
+exports.createMaster = async function (args, res, next, incomingObj) {
+    var Inspection = mongoose.model('Inspection');
+    var inpsection = new Inspection();
+
+    inpsection._schemaName = 'Inspection';
+    incomingObj._epicProjectId && ObjectId.isValid(incomingObj._epicProjectId) && (inpsection._epicProjectId = new ObjectId(incomingObj._epicProjectId));
+    incomingObj._sourceRefId && ObjectId.isValid(incomingObj._sourceRefId) && (inpsection._sourceRefId = new ObjectId(incomingObj._sourceRefId));
+    incomingObj._epicMilestoneId && ObjectId.isValid(incomingObj._epicMilestoneId) && (inpsection._epicMilestoneId = new ObjectId(incomingObj._epicMilestoneId));
+
+    incomingObj.recordName && (inpsection.recordName = incomingObj.recordName);
+    inpsection.recordType = 'Inspection';
+    incomingObj.dateIssued && (inpsection.dateIssued = incomingObj.dateIssued);
+    incomingObj.issuingAgency && (inpsection.issuingAgency = incomingObj.issuingAgency);
+    incomingObj.author && (inpsection.author = incomingObj.author);
+    incomingObj.legislation && (inpsection.legislation = incomingObj.legislation);
+    incomingObj.issuedTo && (inpsection.issuedTo = incomingObj.issuedTo);
+    incomingObj.projectName && (inpsection.projectName = incomingObj.projectName);
+    incomingObj.location && (inpsection.location = incomingObj.location);
+    incomingObj.centroid && (inpsection.centroid = incomingObj.centroid);
+    incomingObj.outcomeStatus && (inpsection.outcomeStatus = incomingObj.outcomeStatus);
+    incomingObj.outcomeDescription && (inpsection.outcomeDescription = incomingObj.outcomeDescription);
+
+    inpsection.dateAdded = new Date();
+    inpsection.publishedBy = args.swagger.params.auth_payload.displayName;
+
+    incomingObj.sourceDateAdded && (inpsection.sourceDateAdded = incomingObj.sourceDateAdded);
+    incomingObj.sourceDateUpdated && (inpsection.sourceDateUpdated = incomingObj.sourceDateUpdated);
+    incomingObj.sourceSystemRef && (inpsection.sourceSystemRef = incomingObj.sourceSystemRef);
+
+    inpsection.read = ['sysadmin'];
+    inpsection.write = ['sysadmin'];
+
+    let savedInspection = null;
+    try {
+        savedInspection = await inpsection.save();
+    } catch (e) {
```

Comment:
I'd prefer 'error' or at least 'err' over just 'e'

Helps with readability, and also makes searching better.  Its pretty impractical to search for 'e', if you were trying to find places where errors are handled and weren't familiar with the code base.

Reply:
You could always search for "catch ("